File: georocket-server/src/main/java/io/georocket/index/generic/DefaultMetaIndexer.java
package io.georocket.index.generic;

import java.util.HashMap;
import java.util.Map;

import io.georocket.index.xml.MetaIndexer;
import io.georocket.storage.ChunkMeta;
import io.georocket.storage.IndexMeta;

/**
 * Default implementation of {@link MetaIndexer} that extracts generic
 * attributes from chunk metadata and adds it to the index.
 * @author Michel Kraemer
 */
public class DefaultMetaIndexer implements MetaIndexer {
  private final Map<String, Object> result = new HashMap<>();

  @Override
  public Map<String, Object> getResult() {
    return result;
  }

  @Override
  public void onIndexChunk(String path, ChunkMeta chunkMeta,
      IndexMeta indexMeta) {
    result.put("path", path);
    result.put("chunkMeta", chunkMeta.toJsonObject());
    if (indexMeta.getTags() != null) {
      result.put("tags", indexMeta.getTags());
    }
    if (indexMeta.getProperties() != null) {
      result.put("props", indexMeta.getProperties());
    }
    if (indexMeta.getCorrelationId() != null) {
      result.put("correlationId", indexMeta.getCorrelationId());
    }
  }
}

File: georocket-server/src/main/kotlin/io/georocket/index/generic/DefaultMetaIndexer.kt
package io.georocket.index.generic

import io.georocket.index.MetaIndexer
import io.georocket.storage.ChunkMeta
import io.georocket.storage.IndexMeta

/**
 * Default implementation of an [MetaIndexer] that extracts generic
 * attributes from chunk metadata and adds it to the index.
 * @author Michel Kraemer
 */
class DefaultMetaIndexer : MetaIndexer {
  override fun onChunk(path: String, chunkMeta: ChunkMeta, indexMeta: IndexMeta): Map<String, Any> {
    val result = mutableMapOf<String, Any>()

    result["path"] = path
    result["chunkMeta"] = chunkMeta.toJsonObject()

    if (indexMeta.tags != null) {
      result["tags"] = indexMeta.tags
    }

    if (indexMeta.properties != null) {
      result["props"] = indexMeta.properties
    }

    if (indexMeta.correlationId != null) {
      result["correlationId"] = indexMeta.correlationId
    }

    return result
  }
}
